Chore: Configure go-ruleguard via golangci-lint

aknuds1 opened the pull request:
Configure golangci-lint to use [go-ruleguard](https://github.com/quasilyte/go-ruleguard) with some initial custom rules (copied from Damian Gryski's [semgrep-go](https://github.com/dgryski/semgrep-go) repo) to catch some Go anti-patterns.

aknuds1 committed:
Chore: Configure go-ruleguard via golangci-lint

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Chore: Configure go-ruleguard via golangci-lint

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

sakjur commented on pkg/bus/bus.go:
```diff
@@ -143,9 +143,9 @@ func (b *InProcBus) Publish(msg Msg) error {
-		if err != nil {
-			return err.(error)
+		err, ok := ret[0].Interface().(error)
+		if ok && err != nil {
+			return err
 		}
```
Thought, maybe it's a good thing to safe up by making sure that if it's _not_ an error we complain

```suggestion
		if ok && err != nil {
			return err
		} else if !ok {
			return fmt.Errorf("expected listener to return an error, got '%v'", reflect.TypeOf(ret[0].Interface()))
		}
```

sakjur commented on pkg/middleware/auth_proxy.go:
```diff
@@ -15,11 +15,11 @@ var header = setting.AuthProxyHeaderName
-	id, err := auth.Login(logger, ignoreCache)
-	if err != nil {
-		logger.Error("Failed to login", "username", username, "message", err.Error(), "error", err.DetailsError,
+	id, e := auth.Login(logger, ignoreCache)
```
In case other reviewers wonder about this:
I looked this up, `e` is an `*auth.Error`, not `error`.

aknuds1 replied:
Yes, I suspect we should convert this code later to return an `error` implementation :)

sakjur replied:
Yeah, I think that's out of scope though :)

aknuds1 replied:
@sakjur Absolutely :D

sakjur commented on pkg/ruleguard.rules.go:
```diff
@@ -0,0 +1,488 @@
+// +build ignore
+
+package gorules
```
Are the commented out code something we've done or from the source?

Are we properly crediting the source of this rule file?

aknuds1 replied:
@sakjur Should we add a comment documenting the origin of the file?

aknuds1 replied:
Added a comment crediting origin.

sakjur reviewed (commented):
Oh, cool, I didn't know that existed.
Are we crediting the original creator of these rules appropriately? I like the direction we're heading in here :+1:

papagian reviewed (approved):
LGTM

aknuds1 commented:
> Oh, cool, I didn't know that existed.
> Are we crediting the original creator of these rules appropriately? I like the direction we're heading in here +1

@sakjur Good question - there's no license, should we do anything but perhaps credit Damian in the commit message?

aknuds1 committed:
Update pkg/bus/bus.go

Co-authored-by: Emil Tullstedt <emil.tullstedt@grafana.com>

aknuds1 committed:
Update pkg/bus/bus.go

Co-authored-by: Emil Tullstedt <emil.tullstedt@grafana.com>

sakjur commented:
@aknuds1 Build fails because we need an import "fmt" as well

aknuds1 committed:
Fix compilation failure

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Fix error check

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Fix error check

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Credit dgryski

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

sakjur reviewed (approved):
Still not sure how we should do attribution but :+1:

aknuds1 commented:
> Still not sure how we should do attribution but +1

@sakjur Did you see the comment I added in pkg/ruleguard.rules.go?

sakjur commented:
Now I did, that looks like a good attribution :+1:
It'd be nice with explicit licensing as well to make sure we're in the clear, but it isn't being linked into the binary, so that's good.

aknuds1 commented:
> Now I did, that looks like a good attribution +1
> It'd be nice with explicit licensing as well to make sure we're in the clear, but it isn't being linked into the binary, so that's good.

@sakjur What can we do about licensing though? I couldn't find any license in the repo.

sakjur commented:
We could ask upstream for a license, but as there are a few external contributors (IANAL but many of them seem to be fortunately not of consequence, spellfixes, minor changes which cannot be considered to be works in and of themselves) it's not necessarily easily done.

aknuds1 commented:
I commented in an existing [issue](https://github.com/dgryski/semgrep-go/issues/16) (made 1 hour ago!) requesting the license.

dgryski commented:
A quick comment here about gating builds/merges on ruleguard failures:  I (on purpoose!) have not been so careful to avoid false positives in the rules I'm creating.  While I'd like to avoid the very obvious ones, some limitations of ruleguard/semgrep prevent full analysis required to *only* catch true issues.  Things like staticcheck and vet have a much higher bar.

aknuds1 commented:
> A quick comment here about gating builds/merges on ruleguard failures: I (on purpoose!) have not been so careful to avoid false positives in the rules I'm creating. While I'd like to avoid the very obvious ones, some limitations of ruleguard/semgrep prevent full analysis required to _only_ catch true issues. Things like staticcheck and vet have a much higher bar.

@dgryski Interesting, thanks for sharing that info. I'm so new to go-ruleguard/semgrep that I'm unsure how they relate to other tools, e.g. staticcheck, vet. Guess I should dig into those once I have more time. Do you have any tips for learning about the alternatives available when it comes to Go static analysis?

dgryski commented:
Basically, staticcheck and vet should have virtually *no* false positives.  However, this means that they can't be used to flag suspicious code which is technically correct.  Also, adding custom checks is compicated.  ruleguard and semgrep fill this void by having a simpler format for writing checks, but it's also a less powerful query language meaning it's hard if not impossible to eliminate classes of false positives that would be possible to eliminate if you were writing a check with the full power of Go.  ruleguard and semgrep are trading simplicity of writing custom checks for accuracy.

aknuds1 commented:
@dgryski Thanks for clarifying that! There's clearly a lot for me to dig into :)

marefr reviewed (approved):
Looks useful :+1: LGTM

dprokop committed:
Bump rxjs to 6.6.3 (#28657)

chugunov committed:
Docker:  use root group in the custom Dockerfile (#28639)

wbrowne committed:
Plugins: Fix descendent frontend plugin signature validation (#28638)

* move plugin root check to earlier in validation process

* remove comment

* only check root if necessary

Commit:
Docs: Describe pipeline aggregation changes in v7.3 (#28660)

dprokop committed:
Add info about CSV download for Excel in What's new article (#28661)

* Add info about CSV download for Excel in What's new article

* Update docs/sources/whatsnew/whats-new-in-v7-3.md

* Update docs/sources/whatsnew/whats-new-in-v7-3.md

zoltanbedi committed:
Cloudwatch: Fix duplicate metric data (#28642)

* Cloudwatch: Fix duplicate metric data

* Refactor reduce function to for of

torkelo committed:
PanelMenu: Fixes panel submenu not being accessible for panels close to the right edge of the screen (#28666)

* Dropdowns: Trying to fix dropdown menus

* Dropdowns: Trying to fix dropdown menus

* removed now unnessary wrapper ref

* Upodates

* Remove export

torkelo committed:
TextPanel: Fixes problems where text panel would show old content (#28643)

zoltanbedi committed:
Chore: allow higher nodejs version than 12 (#28624)

ivanahuckova committed:
Loki: Fix error when some queries return zero results (#28645)

* Fix logSeriesToLogsModel when some data frames have empty fields

* Update test

* Update public/app/core/logs_model.ts

Co-authored-by: Zoltán Bedi <zoltan.bedi@gmail.com>

Co-authored-by: Zoltán Bedi <zoltan.bedi@gmail.com>

aocenas committed:
Prometheus: Fix copy paste behaving as cut and paste (#28622)

mckn committed:
Build: changing docs docker image to prevent setting up frontend devenv. (#28670)

* Build: changing docs dev-environment

I have added a new image that will include the auto-generated frontend docs so you don't have to generate those locally prior to running the doc container. This has a dependency on this PR: https://github.com/grafana/website/pull/2744 and that we publish a version of the `grafana/docs-devenv` image.

* will ignore everything (even symbolic links)

* fixed spelling.

* added newline.

* updated image name.

aknuds1 committed:
Drone: Don't build on Windows for PRs (#28663)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

dprokop committed:
Changelog update - 7.3.1 (#28699)

* Changelog update

* Update CHANGELOG.md

* Update CHANGELOG.md

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

dprokop committed:
Update latest version to 7.3.1 (#28701)

aocenas committed:
Chore: Add some scenario tests for Explore (#28534)

* Add tests for main scenarios

* Add tests for split

* Remove unnecessary code

* Fix opening datasource picker

* Mock missing function

achatterjee-grafana committed:
updates from https://github.com/grafana/grafana/pull/28679 (#28708)

ceh committed:
Chore: Convert sqlstore annotation test from GoConvey to testify (#28715)

* Chore: Convert sqlstore annotation test from GoConvey to testify

* Use require.NoError

* Use assert.Empty

* Use assert.NoError in test cleanup

eamonryan committed:
Fix typo in unsigned plugin warning (#28709)

Typo fix!

hugohaggmark committed:
Templating: Speeds up certain variable queries for Postgres, MySql and MSSql (#28686)

hugohaggmark committed:
TableFilters: Fixes filtering with field overrides (#28690)

* TableFilters: Fixes filtering with field overrides

* Refactor: changes after PR comments

aknuds1 committed:
Drone: Upgrade build-pipeline tool (#28728)

* Drone: Upgrade build-pipeline tool

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Drone: Raise Drone's memory limit for build-storybook

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Chore: Pin nginx base image in nginx proxy Dockerfiles (#28730)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

Clarity-89 committed:
AlertRuleList: Add keys to alert rule items (#28735)

torkelo committed:
Dependencies: Remove unused dependency (#28711)

* Dependencies: Remove unused dependency

* Remove import

* Updated

jackw committed:
Migration: Remove LegacyForms from dashboard folder permissions (#28564)

* refactor(permissionlist): replace legacyform components with next-gen equivalents

* refactor(permissionlist): replace legacyform components

* refactor(permissionlist): clean up PermissionsInfo

* feat(grafana-ui): define line-height for Select option descriptions

* refactor(permissionlist): remove debug code

* refactor(permissionlist): simplify component tree with horizontalgroup. Add label

* refactor(permissionlist): remove redundant classname from Select

dprokop committed:
Plugin page: Fix dom validation warning (#28737)

tabarra committed:
LoginPage: Removed auto-capitalization from the login form (#28716)

* Removed auto-capitalization from the login form

On mobile browsers, by default most keyboards will auto capitalize the first letter of the email/username, which is case sensitive.  
This PR fixes that.

* Fixed autoCapitalize react keyword

torkelo committed:
Provisioning: Fixed problem with getting started panel being added to custom home dashboard (#28750)

* Provisioning: Fixed problem with getting started panel being added to custom home dashboard

* Fixed comment

* Update pkg/api/dashboard.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

jackw committed:
Dashboard: fix view panel mode for Safari / iOS (#28702)

* fix(dashboard): flex-basis so child height percentages are respected

* fix(dashboard): reset scrollTop when entering view panel mode

ivanahuckova committed:
Explore: Remove redundant decodeURI and fix urls (#28697)

* Remove redundant path URIdecoding

* Remove redundant comments

aknuds1 committed:
gtime: Add ParseDuration function (#28525)

* gtime: Make calculations independent of current time

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Introduce gtime.ParseDuration function

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* gtime: Fix ParseDuration

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

achatterjee-grafana committed:
Reoeragnization. (#28760)

peterholmberg committed:
Docs: Add docs for InfoBox component (#28705)

* add mdx

* fixes after pr feedback

aknuds1 committed:
devenv: Upgrade MSSQL Docker image (#28749)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Drone: Upgrade build pipeline tool (#28769)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

dprokop committed:
grafana/toolkit: Extract CHANGELOG when building plugin (#28773)

torkelo committed:
Cloudwatch: Fix issue with field calculation transform not working properly with Cloudwatch data (#28761)

* Cloudwatch: Fix issue with reducer transform not working properly with Cloudwatch data

* Updated go sdk and updated code

* CloudWatch: Improve test

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

ndeevy committed:
Docs: Made usage of config/configuration consistent #19270 (#28167)

Rebased ndeevy:docs-config/configuration-consistency-update from grafana/master to resolve conflicts

Co-authored-by: ndeevy <ndeevy@redhat.com>

zen committed:
Docs: Add NGINX example for using websockets to Loki (#27998)

* feat: added NGINX example for using websockets to Loki

* chore: corrections after review

* chore: more correrctions from review

* chore: creview corrections

* chore: addressed review comment by @achatterjee-grafana

peterholmberg committed:
ColorSchemes: Add new color scheme (#28719)

torkelo committed:
EventBus: Introduces new event bus with emitter backward compatible interface (#27564)

* updated

* Experimenting with event bus with legacy support

* Before switch to emitter

* EventBus & Emitter unification

* Everything using new EventBus

* Making progress

* Fixing merge issues

* Final merge issues

* Updated

* Updates

* Fix

* Updated

* Update

* Update

* Rename methods to publish and subscribe

* Ts fixes

* Updated

* updated

* fixing doc warnigns

* removed unused file

torkelo committed:
Dependencies: Update angularjs to 1.8.2 (#28736)

* Angular: Update to angukar v1.8 with backward compatability monkey patches

* Update fix

* Updated

* Updated

* Fixes

* Update wording

* Update

aknuds1 committed:
Plugin Dockerfiles: Upgrade Go, golangci-lint, gcloud SDK (#28767)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

mckn committed:
Docs: added code comments to frontend packages. (#28784)

* fixed docs linting errors.

* decreased the limit.

torkelo committed:
Events: Remove unused or unnecessary events  (#28783)

* Events: Removing unused or unnessary events

* More cleanup

* Updated

ryantxu committed:
Live: improve broadcast semantics and avoid double posting (#28765)

dprokop committed:
Adds storybook integrity check to drone config (#28785)

* Adds storybook integrity check to dron config

* Update starlark files

* Review comments

* Bump grabpl version

* Drone: Fix dangling dependency

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

torkelo committed:
Graph: Fixes event emit function error (#28795)

hugohaggmark committed:
Variables: Fixes URL values for dependent variables (#28798)

dprokop committed:
grafana/toolkit: Drop console and debugger statements by default when building plugin with toolkit (#28776)

dprokop committed:
Panel inspect: Interpolate variables in panel inspect title (#28779)

* Interpolate variables in panel inspect title

* Update public/app/features/dashboard/components/Inspector/InspectContent.tsx

Co-authored-by: Torkel Ödegaard <torkel@grafana.org>

* fix typo

Co-authored-by: Torkel Ödegaard <torkel@grafana.org>

yenalsnavaj committed:
Variables: Adds named capture groups to variable regex (#28625)

* Dashboard: Add named capture groups to variable query regex

Variable query regex are able to use 'text' and 'value' named capture
groups to allow for separate display text to be extracted from the
query result. e.g.

Using a regex of /foo="(?<text>[^"]+)|bar="(?<value>[^"]+)/g on a query
result of metric{foo="FOO", bar="BAR"} would result in the variable
value being set to 'BAR' but display text being set to 'FOO'

Resolves #21076

* Improve regex capture group documentation

* Update docs/sources/variables/filter-variables-with-regex.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

* Apply suggestions from code review

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

* Apply suggestions from code review

Co-authored-by: Hugo Häggmark <hugo.haggmark@gmail.com>

* Use text capture if value capture does not match

This is to keep the behaviour consistent with the current behavior. See
discussion https://github.com/grafana/grafana/pull/28625/files#r516490942

* Improve regex field placeholder and tooltip message

To make the feature more discoverable to users the place holder example
now includes the named capture groups. The tool tip message also
includes a reference and link to the documentation.

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>
Co-authored-by: Hugo Häggmark <hugo.haggmark@gmail.com>

triptu committed:
Dashboard: Add Datetime local (No date if today) option in panel axes' units (#28011)

* Add Datetime local (No date if today) option

* Add more tests

* Revert to using function for local dateTime format
This is required as Intl api used to convert date to local format is not present in
node environments

jedstar committed:
StatPanels: Add new calculation option for percentage difference  (#26369)

* Update fieldReducer.ts

addition of percentage difference to the singlestat panel

* Update time_series2.ts

* Update module.ts

* Update calculations-list.md

* Update docs/sources/panels/calculations-list.md

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

* Update public/app/plugins/panel/singlestat/module.ts

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

* Update module.ts

* Update calculations-list.md

* Update calculations-list.md

* Update module.ts

* Update fieldReducer.ts

* Update fieldReducer.ts

* Update fieldReducer.test.ts

* change name to remove wildcard characters

* Update calculations-list.md

* Update time_series2.ts

Fix spelling

* Update module.ts

* Update fieldReducer.ts

* formatting

* Update fieldReducer.ts

* Update fieldReducer.test.ts

* Update fieldReducer.test.ts

Co-authored-by: Diana Payton <52059945+oddlittlebird@users.noreply.github.com>

dprokop committed:
Remove homelinks panel (#28808)

ivanahuckova committed:
Icon: Replace font awesome icons where possible (#28757)

* Replace font awesome icons where possible

* Implement small updates

dprokop committed:
core and grafana/toolkit: Use latest version of grafana-eslint-conifg (#28816)

dprokop committed:
grafana/toolkit: allow builds with lint warnings (#28810)

* grafana/toolkit: allow builds with lint warnings

* Remove redundant code

torkelo committed:
Gauge: Improve font size auto sizing (#28797)

* Gauge: Improved font size calculations

* Added some comments

* update

* Moving to variable

wbrowne committed:
increase blob column size for encrypted dashboard data (#28831)

torkelo committed:
Alert: Remove z-index on Alert component so that it does not overlay ontop of other content (#28834)

dslizardo committed:
Currency: add Philippine peso currency (PHP) (#28823)

torkelo committed:
Marked: Upgrade and always sanitize by default (#28796)

* Marked: Upgrade and always sanitize by default

* Added test

* corrected text panel logic

hugohaggmark committed:
Variables: Adds variables inspection (#25214)

* Chore: initial commit

* Refactor: adds usages

* Refactor: adds FeatureInfoBox

* Refactor: uses graphs instead

* Refactor: adds id, name, title to prop instead of key

* Refactor: adds usages to VariablesList

* Refactor: Moves unknowns to first page

* Refactor: minor stylings and icons

* Refactor: styling

* Refactor: changed to button with modal

* Refactor: adds VariablesDependenciesButton

* Refactor: changes after UX feedback

* Refactor: renames heading

* Refactor: changes after PR comments

* Refactor: small changes after PR comments

mckn committed:
Units: added support to handle negative fractional numbers. (#28849)

aknuds1 committed:
Chore: Remove dead code (#28664)

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Chore: Fix issues found by staticcheck (#28802)

* Fix linting issues

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

zoltanbedi committed:
Prometheus: fix missing labels from value (#28842)

zoltanbedi committed:
Tempo: fix test data source (#28836)

Clarity-89 committed:
Disable selecting enterprise plugins with no license (#28758)

* Add unlicensed property to plugins

* Disable selecting unlicensed plugin

* Add customizable plugin market place url

* License: workaround enabled only in enterprise

* linter

* Move licensing info to front end

* Update pkg/services/licensing/oss.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update pkg/services/licensing/oss.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update pkg/setting/setting.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update pkg/setting/setting.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update pkg/api/frontendsettings.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update sample.ini

* Update docs

* Update packages/grafana-runtime/src/config.ts

Co-authored-by: Torkel Ödegaard <torkel@grafana.org>

* Update public/app/features/datasources/state/buildCategories.ts

Co-authored-by: Torkel Ödegaard <torkel@grafana.org>

* Update pkg/api/frontendsettings.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Update pkg/setting/setting.go

Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix spelling

Co-authored-by: Leonard Gram <leo@xlson.com>
Co-authored-by: Arve Knudsen <arve.knudsen@gmail.com>
Co-authored-by: Torkel Ödegaard <torkel@grafana.org>

aknuds1 committed:
Chore: Fix staticcheck issues (#28854)

* Chore: Fix issues reported by staticcheck

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Undo changes

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Chore: Fix staticcheck issues (#28860)

* Chore: Fix issues reported by staticcheck

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Undo changes

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Chore: Fix issues reported by staticcheck

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix test

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

* Fix test

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/ruleguard

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/ruleguard

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/ruleguard

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/ruleguard

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/ruleguard

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/ruleguard

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/ruleguard

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/ruleguard

aknuds1 committed:
Fix linting issue

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/ruleguard

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/ruleguard

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/ruleguard

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/ruleguard

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/ruleguard

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/ruleguard

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/ruleguard

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/ruleguard

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/ruleguard

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/ruleguard

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/ruleguard

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/ruleguard

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/ruleguard

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/ruleguard

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/ruleguard

aknuds1 committed:
Merge remote-tracking branch 'origin/master' into chore/ruleguard

aknuds1 committed:
Include semgrep-go's license

Signed-off-by: Arve Knudsen <arve.knudsen@gmail.com>
